File: exercises/java/src/main/java/tax/DefaultTaxCalculator.java
package tax;

public class DefaultTaxCalculator extends TaxCalculator {

    @Override
    int calculateTax(Vehicle vehicle) {
        int co2Emissions = vehicle.getCo2Emissions();
        if (co2Emissions == 0) {
            return 0;
        }
        if (isBetweenBounds(co2Emissions, 1, 50)) {
            return 10;
        }
        if (isBetweenBounds(co2Emissions, 51, 75)) {
            return 25;
        }
        if (isBetweenBounds(co2Emissions, 76, 90)) {
            return 105;
        }
        if (isBetweenBounds(co2Emissions, 91, 100)) {
            return 125;
        }
        if (isBetweenBounds(co2Emissions, 101, 110)) {
            return 145;
        }
        if (isBetweenBounds(co2Emissions, 111, 130)) {
            return 165;
        }
        if (isBetweenBounds(co2Emissions, 131, 150)) {
            return 205;
        }
        if (isBetweenBounds(co2Emissions, 151, 170)) {
            return 515;
        }
        if (isBetweenBounds(co2Emissions, 171, 190)) {
            return 830;
        }
        if (isBetweenBounds(co2Emissions, 191, 225)) {
            return 1240;
        }
        if (isBetweenBounds(co2Emissions, 226, 255)) {
            return 1760;
        }
        return 2070;
    }

    private boolean isBetweenBounds(int value, int lowerBound, int upperBound) {
        return value >= lowerBound && value <= upperBound;
    }
}

File: exercises/kotlin/src/main/kotlin/tax/DefaultTaxCalculator.kt
package tax

class DefaultTaxCalculator : TaxCalculator() {

    override fun calculateTax(vehicle: Vehicle?): Int {
        val co2Emissions = vehicle?.co2Emissions ?: throw NullPointerException()

        return when (co2Emissions) {
            0 -> 0
            in 1..50 -> 10
            in 51..75 -> 25
            in 76..90 -> 105
            in 91..100 -> 125
            in 101..110 -> 145
            in 111..130 -> 165
            in 131..150 -> 205
            in 151..170 -> 515
            in 171..190 -> 830
            in 191..225 -> 1240
            in 226..255 -> 1760
            else -> 2070
        }
    }
}